jupyterlab_gcedetails/src/components/hardware_scaling_status.tsx

```diff
@@ -0,0 +1,173 @@
+import * as React from 'react';
+import { stylesheet } from 'typestyle';
+import LinearProgress from '@material-ui/core/LinearProgress';
+import { createStyles, withStyles, Theme } from '@material-ui/core/styles';
+import Button from '@material-ui/core/Button';
+import { ClientTransportService } from 'gcp_jupyterlab_shared';
+import { HardwareConfiguration, Details } from '../data';
+import { NotebooksService } from '../service/notebooks_service';
+import { AuthTokenRetrieval } from './auth_token_retrieval';
+
+const BorderLinearProgress = withStyles((theme: Theme) =>
+  createStyles({
+    root: {
+      width: 400,
+      height: 15,
+      borderRadius: 5,
+    },
+    colorPrimary: {
+      backgroundColor:
+        theme.palette.grey[theme.palette.type === 'light' ? 200 : 700],
+    },
+    bar: {
+      backgroundColor: '#1a90ff',
+      borderRadius: 5,
+    },
+  })
+)(LinearProgress);
+
+const STYLES = stylesheet({
+  flexContainer: {
+    width: 500,
+    height: 300,
+    display: 'flex',
+    flexDirection: 'column',
+    justifyContent: 'center',
+    alignItems: 'center',
+  },
+  heading: {
+    fontSize: '24px',
+  },
+  paragraph: {
+    height: 60,
+    width: 350,
+    textAlign: 'center',
+  },
+});
+
+enum Status {
+  'Authorizing' = 0,
+  'Stopping Instance' = 1,
+  'Reshaping Instance' = 2,
+  'Starting Instance' = 3,
+  'Complete' = 4,
+  'Error' = 5,
+}
+
+const statusInfo = [
+  'Getting authorization token to reshape machine. A dialog box should have popped up.',
```

Comment:
What do you think about "Please complete the OAuth 2.0 authorization steps in the popup" for the second sentence?